### PR TITLE
Fix failing firebase_test_lab_test, fix dart analyzer warnings

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/46041.
https://github.com/flutter/flutter/issues/46041 increased the timeout in the command but not the hard-coded one in the tests.

Also fix all Dart analyzer warnings.

### CODE REVIEW

File: test/publish_plugin_command_test.dart [hunk 1]
```diff
@@ -59,7 +60,7 @@ void main() {
   group('Initial validation', () {
     test('requires a package flag', () async {
       await expectLater(() => commandRunner.run(<String>['publish-plugin']),
-          throwsA(const isInstanceOf<ToolExit>()));
```
Comment (jmagman) [inline test/publish_plugin_command_test.dart]: Deprecated.

File: test/util.dart [hunk 2]
```diff
@@ -172,16 +173,27 @@ class ProcessCall {
...
         listsEqual(args, otherCall.args) &&
         workingDir == otherCall.workingDir;
   }
 
+  @override
+  int get hashCode =>
+      executable?.hashCode ??
+      0 ^ args?.hashCode ??
+      0 ^ workingDir?.hashCode ??
+      0;
```
Comment (jmagman) [inline test/util.dart]: dartfmt is doing something interesting here.  I can split it up if you want.

File: analysis_options.yaml [hunk 3]
```diff
@@ -19,8 +19,6 @@
 # indicating the # of violations for that rule.
 
 analyzer:
-  language:
-    enableSuperMixins: true
```
[the remark this answers is not in the document]
Comment (mklim) [reply analysis_options.yaml]: I don't want to turn this PR into more of a yak shave than it already is, but WDYT about changing this to use https://pub.dev/packages/pedantic instead like flutter/plugins has been switched to? That should keep us from using out of date rules like this, and other rules that are really more suitable for the framework. Totally understand if you'd rather not go that far in this patch though.

File: test/util.dart (hunk 2 above)
Comment (mklim) [reply test/util.dart]: :man_shrugging: this is fine.

**Comment by jmagman** [comment]
> The pubspec.yaml and CHANGELOG.md should be updated too. I think I may have missed that in the last PR also, sorry. 

Ugh my bad...
https://github.com/flutter/plugin_tools/pull/72

File: analysis_options.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmagman) [reply analysis_options.yaml]: Good idea. I did not include the `public_member_api_docs` rule since we don't need to document public APIs.

**Comment by jmagman** [comment]
> It looks like even the lints that were here aren't being tested in this package's CI. We should probably file an issue to actually analyze this package in the presubmits too.

https://github.com/flutter/flutter/issues/46042

**Pull request merged**